frontend/src/metabase/admin/permissions/components/PermissionsGrid.jsx

```diff
@@ -210,9 +231,14 @@ class GroupPermissionCell extends Component {
                                     postAction: permission.postAction
                                 })
                             }
-                            let confirmText = permission.confirm && permission.confirm(group.id, entity.id, value);
-                            if (confirmText) {
-                                this.setState({ confirmText, confirmAction });
+                            let confirmations = permission.confirm && permission.confirm(group.id, entity.id, value);
+                            if (Array.isArray(confirmations)) {
+                                confirmations = confirmations.filter(c => c);
+                            } else if (confirmations) {
+                                confirmations = [confirmations];
+                            }
```

Comment:
Re style -- Why can confirmations be an array or a single entry? This seems more error prone than having a single type for confirmations.